src/openrct2/CmdlineSprite.cpp

```diff
@@ -708,66 +696,44 @@ int32_t cmdline_for_sprite(const char** argv, int32_t argc)
 
         fprintf(stdout, "Building: %s\n", spriteFilePath);
 
-        size_t i;
-        json_t* sprite_description;
+        json_t sprite_description;
 
-        json_array_foreach(sprite_list, i, sprite_description)
+        for (auto& [jsonKey, jsonSprite] : jsonSprites.items())
         {
-            if (!json_is_object(sprite_description))
+            if (!jsonSprite.is_object())
             {
-                fprintf(stderr, "Error: expected object for sprite %lu\n", static_cast<unsigned long>(i));
-                json_decref(sprite_list);
+                fprintf(stderr, "Error: expected object for sprite %s\n", jsonKey.c_str());
                 return -1;
             }
 
-            json_t* path = json_object_get(sprite_description, "path");
-            if (!json_is_string(path))
+            json_t path = jsonSprite["path"];
+            if (!path.is_string())
             {
-                fprintf(stderr, "Error: no path provided for sprite %lu\n", static_cast<unsigned long>(i));
-                json_decref(sprite_list);
+                fprintf(stderr, "Error: no path provided for sprite %s\n", jsonKey.c_str());
                 return -1;
             }
-            // Get x and y offsets, if present
-            json_t* x_offset = json_object_get(sprite_description, "x_offset");
-            json_t* y_offset = json_object_get(sprite_description, "y_offset");
-
-            // Get palette option, if present
-            bool keep_palette = false;
-            json_t* palette = json_object_get(sprite_description, "palette");
-            if (json_is_string(palette))
-            {
-                const char* option = json_string_value(palette);
-                if (strncmp(option, "keep", 4) == 0)
-                {
-                    keep_palette = true;
-                }
-            }
+            std::string strPath = Json::GetString(path);
 
-            // Get forcebmp option, if present
-            bool forceBmp = false;
-            json_t* forceBmpObject = json_object_get(sprite_description, "forceBmp");
-            if (palette && json_is_boolean(forceBmpObject))
-            {
-                forceBmp = json_boolean_value(forceBmpObject);
-            }
+            json_t x_offset = jsonSprite["x_offset"];
+            json_t y_offset = jsonSprite["y_offset"];
+
+            bool keep_palette = Json::GetString(jsonSprite["palette"]) == "keep";
+            bool forceBmp = Json::GetBoolean(jsonSprite["forceBmp"]);
```

Comment:
This used to be set only if `jsonSprite["palette"]` was set, which is not the case anymore.